shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -121,13 +122,15 @@ public void writeFields(DataOutput out) throws IOException {
             out.writeLong(requestId);
             out.writeUTF(segment);
             out.writeUTF(correctHost);
+            out.writeUTF(serverStackTrace);
         }
 
         public static WireCommand readFrom(DataInput in, int length) throws IOException {
             long requestId = in.readLong();
             String segment = in.readUTF();
             String correctHost = in.readUTF();
-            return new WrongHost(requestId, segment, correctHost);
+            String serverStackTrace = in.readUTF();
```

Comment:
This way a *new* client won't be able to read a response from an *old* server version. Is this expected ? Or is there some other versioning mechanism?

Reply:
Thanks for your comment @eolivelli. I addressed it by checking whether an error message contains `serverStackTrace` field or not before actually attempting to read it. By doing this, messages from older clients will still be compatible. I also increase the protocol version number to denote the change introduced.